### PR TITLE
IndexFeatureFile: more informative error message when trying to index a malformed file

### LINKED ISSUE #4184: IndexFeatureFile can't index .vcf.gz using ProgressReportingDelegatingCodec
When trying to index a gzipped VCF at `/humgen/gsa-hpprojects/dev/gauthier/reblockGVCF/hybrid.m38.vcf.gz`I get the error `A USER ERROR has occurred: This tool does not supports indexing of block-compressed files for ProgressReportingDelegatingCodec`

Indexing this file with tabix is speedy and successful.

### PR DESCRIPTION
**Pull request by droazen**
Resolves #4184

### CODE REVIEW

**Comment by droazen** [comment]
@ldgauthier please review

File: src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java [hunk 1]
```diff
@@ -169,7 +169,7 @@ public void testVCFGZIndex_inferredName(){
         checkIndex(index, Arrays.asList("1", "2", "3", "4"));
     }
 
-    @Test(expectedExceptions = UserException.MalformedFile.class)
```
Comment (ldgauthier) [inline src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: @lbergelson and I have lamented how many of our tests expect non-subclassed UserExceptions and then get called with missing input files and everything passes.  I'd be happier if IndexFeatureFile threw a more specific UserException, but I understand you're lumping together a few different cases.  Maybe add a UserException.TribbleParseError or something similarly named?

Comment (droazen) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: Unfortunately the underlying cause here can be anything from "file is malformed" to "codec does not support tabix", and it all comes out of htsjdk as a `TribbleException`. I thought about using `UserException.MalformedFile` here, which will be accurate 90%+ of the time, but there is that other 10%...

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: Over time every test that catches raw UserException breaks and starts to pass incorrectly due to changes in the command line inputs.

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: Yeah, it's hard when all we know is "Something is wrong"

Comment (droazen) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: But I can switch to `UserException.MalformedFile` all the same, if you prefer.

Comment (ldgauthier) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: I would prefer UserException.MalformedFile to vanilla UserException.

Comment (droazen) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: @lbergelson Again, the problem is in htsjdk in lumping all tribble errors together. Our choices in GATK are to throw a more-specific but sometimes-completely-inaccurate exception, or a more general exception with exposure of the underlying error message.

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: Fixing htsjdk errors would be a nice benefit of redoing things in htsjdk, although it's going to be a big undertaking I think.

Comment (droazen) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: How about I introduce a `UserException.FailureToIndexFile` exception here -- that would be accurate and address your concerns about tests erroneously passing.

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: I'm actually more in favor of throwing the more general exception with the exposure of the underlying issue.  It's just nasty for testing.

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: Good idea!

Comment (ldgauthier) [reply src/test/java/org/broadinstitute/hellbender/tools/IndexFeatureFileIntegrationTest.java]: I like it.

**Comment by droazen** [comment]
@ldgauthier Introduced a suitably-vague `UserException.CouldNotIndexFile` in response to your comments. Ready for a last look.

**Comment by ldgauthier** [comment]
:+1:

**Pull request merged**